languages/javalang/builder-renderer/java-builder-client/src/main/kotlin/io/vrap/codegen/languages/javalang/client/builder/requests/JavaRequestBuilderResourceRenderer.kt

```diff
@@ -82,9 +83,13 @@ class JavaRequestBuilderResourceRenderer @Inject constructor(val api: Api, overr
 
     private fun Method.constructorArguments(): String? {
         return if(this.bodies != null && this.bodies.isNotEmpty()){
-            val methodBodyVrapType = this.bodies[0].type.toVrapType() as VrapObjectType
-            val methodBodyArgument = "${methodBodyVrapType.`package`}.${methodBodyVrapType.simpleClassName} ${methodBodyVrapType.simpleClassName.decapitalize()}"
-            methodBodyArgument
+            if(this.bodies[0].type.toVrapType() is VrapObjectType) {
```

Comment:
It would be nice if we could extract the condition so that we don't have to duplicate it below. WDYT?